--- zenmonitor.orig/src/zenmonitor.c    2020-12-22 02:41:39.829243703 +0100
+++ zenmonitor/src/zenmonitor.c 2020-12-22 02:42:42.942690906 +0100
@@ -10,6 +10,7 @@

 #define AMD_STRING "AuthenticAMD"
 #define ZEN_FAMILY 0x17
+#define ZEN3_FAMILY 0x19

 // AMD PPR = https://www.amd.com/system/files/TechDocs/54945_PPR_Family_17h_Models_00h-0Fh.pdf

@@ -31,7 +32,7 @@
     __get_cpuid(1, &eax, &ebx, &ecx, &edx);

     ext_family = ((eax >> 8) & 0xF) + ((eax >> 20) & 0xFF);
-    if (ext_family != ZEN_FAMILY){
+    if (ext_family != ZEN_FAMILY && ext_family != ZEN3_FAMILY){
         return FALSE;
     }

