diff --git a/Makefile b/Makefile
index 8f90462..3563b80 100644
--- a/Makefile
+++ b/Makefile
@@ -3,9 +3,13 @@
 DESTDIR=
 PREFIX=/usr
 
+CONFIG_DIR=/etc/keyd
+LOCK_FILE=/var/lock/keyd.lock
+LOG_FILE=/var/log/keyd.log
+
 VERSION=1.1.2
 GIT_HASH=$(shell git describe --no-match --always --abbrev=40 --dirty)
-CFLAGS=-DVERSION=\"$(VERSION)\" -DGIT_COMMIT_HASH=\"$(GIT_HASH)\"
+CFLAGS=-DVERSION=\"$(VERSION)\" -DGIT_COMMIT_HASH=\"$(GIT_HASH)\" -DLOCK_FILE=\"$(LOCK_FILE)\" -DLOG_FILE=\"$(LOG_FILE)\" -DCONFIG_DIR=\"$(CONFIG_DIR)\"
 
 all:
 	mkdir -p bin
diff --git a/src/config.h b/src/config.h
index a6465b8..3a53043 100644
--- a/src/config.h
+++ b/src/config.h
@@ -29,7 +29,11 @@
 #include "keys.h"
 
 #define MAX_LAYERS 32
+
+#ifndef CONFIG_DIR
 #define CONFIG_DIR "/etc/keyd"
+#endif
+
 #define MAX_LAYER_NAME_LEN 256
 #define MAX_MACRO_SIZE 256
 #define MAX_MACROS 256
diff --git a/src/main.c b/src/main.c
index 86dbc7e..a697170 100644
--- a/src/main.c
+++ b/src/main.c
@@ -42,8 +42,14 @@
 
 #define UINPUT_DEVICE_NAME "keyd virtual keyboard"
 #define MAX_KEYBOARDS 256
+
+#ifndef LOCK_FILE
 #define LOCK_FILE "/var/lock/keyd.lock"
+#endif
+
+#ifndef LOG_FILE
 #define LOG_FILE "/var/log/keyd.log" //Only used when running as a daemon.
+#endif
 
 #ifdef DEBUG
 	#define dbg(fmt, ...) warn("%s:%d: "fmt, __FILE__, __LINE__, ## __VA_ARGS__)
